Sources/SwiftParser/Patterns.swift

```diff
@@ -218,16 +231,19 @@ extension Parser {
           arena: self.arena
         )
       )
-    case (.rhs, let handle)?:
-      let bindingSpecifier = self.eat(handle)
-      let value = self.parseMatchingPattern(context: .bindingIntroducer)
-      return RawPatternSyntax(
-        RawValueBindingPatternSyntax(
-          bindingSpecifier: bindingSpecifier,
-          pattern: value,
-          arena: self.arena
+    case (.rhs(let introducer), let handle)?:
+      var backtrack = lookahead()
+      if backtrack.shouldParsePatternBinding(introducer: introducer) {
```

Comment:
We call it `lookahead` not `backtrack` in the new parser because the parser doesn’t actually backtrack. It clones itself on `lookahead()` and the clone looks ahead while the parser itself stays at the same position.

Reply:
I think all of these uses can be replaced by `withLookahead`, which I didn't know about. I used `backtrack` because it is used in several places to name a lookahead variable elsewhere in the codebase; are those other places needing updating?

```
% git grep 'backtrack =' 
Sources/SwiftParser/Expressions.swift:      var backtrack = self.lookahead()
Sources/SwiftParser/Expressions.swift:      var backtrack = self.lookahead()
Sources/SwiftParser/Expressions.swift:      let backtrack = self.lookahead()
Sources/SwiftParser/Expressions.swift:        var backtrack = lookahead()
Sources/SwiftParser/Expressions.swift:      var backtrack = self.lookahead()
Sources/SwiftParser/Expressions.swift:    var backtrack = self.lookahead()
Sources/SwiftParser/Expressions.swift:    var backtrack = self.lookahead()
Sources/SwiftParser/Lookahead.swift:        var backtrack = self.lookahead()
Sources/SwiftParser/Names.swift:    var backtrack = self.lookahead()
Sources/SwiftParser/Names.swift:      var backtrack = self.lookahead()
Sources/SwiftParser/Patterns.swift:      var backtrack = self.lookahead()
Sources/SwiftParser/Patterns.swift:      var backtrack = lookahead()
Sources/SwiftParser/Types.swift:        var backtrack = self.lookahead()
Sources/SwiftParser/Types.swift:        var backtrack = self.lookahead()
```

Reply:
Huh, yeah, they should probably all be updated. Probably still a leftover from the migration from C++. Filed https://github.com/apple/swift-syntax/issues/2494